### DISCUSSION TITLE
How I can add/use my Neovim commands?

### DISCUSSION THREAD

**Discussion by MannyFay** [vscode-neovim/vscode-neovim#1908] [Q&A]
Like in the documentation explained, I've created a `keybindings.json`.  
This is the very whole content of it:
```json
{
  "command": "editor.action.commentLine",
  // the key sequence to activate the binding
  "key": "cmd+/",
  // don't activate during insert mode
  "when": "editorTextFocus && neovim.mode != insert",
  // the input to send to Neovim
  "args": "<Leader>/"
}
```
I want to use `<Leader>/` to comment in/out a line, but it doesn't work. Is there something wrong with my keybinding? Did I've misunderstood something?

**Comment by MannyFay** [comment]
Gosh! That's a hell of configuration and documentation!

So, in my original Packer plugins file (not updated yet to Lazy) I have to set this:
```lua
if vim.g.vscode then
    -- VSCode extension
else
    -- ordinary Neovim
end
```
...and all my Nvim plugins I want to use in VS Code I list in `if vim.g.vscode` part? All I don't want to use in VSC I put into the else part?

Then, in the documentation is always mentioned the `init.vim` file - man! It is Neovim, so it has to be Lua. If people want to use Vim script, they use Vim instead of Neovim!

So, if I want to use keymaps like in Neovim, I have to put them in my Neovim keymaps file (mixing Nvim configuration with VSC-Nvim configuration?!) or I create a new one only for using in VSC (then I have two places that I have to care of?!)?

Can show me somebody a well written Lua configuration for this extension?
Can somebody give me an example how I exactly implement commenting in/out lines with the Neovim command `<Leader>/` (the VSC action is `editor.action.commentLine`)?

**Comment by justinmk** [comment]
> ...and all my Nvim plugins I want to use in VS Code I list in `if vim.g.vscode` part? All I don't want to use in VSC I put into the else part?

Yes. Or rather, the parts you want to include/exclude conditionally for vscode, go in that if/else block. The rest of your nvim config will "mostly" work. If you find some plugin that doesn't work well, then its configuration should go in the `else` block.

There are numerous examples in the readme showing how to call vscode commands (the same id that you would call in vscode itself): https://github.com/vscode-neovim/vscode-neovim?tab=readme-ov-file#examples

**Comment by MannyFay** [comment]
My leader key is regularly defined at the very top of this `keymaps.lua` file.

![Screenshot 2024-04-25 at 20 27 27](https://github.com/vscode-neovim/vscode-neovim/assets/45234439/efd01620-5e20-403c-8ae4-5e902f4673b1)
![Screenshot 2024-04-25 at 20 27 52](https://github.com/vscode-neovim/vscode-neovim/assets/45234439/adf04a2d-fb8f-473c-a8fb-731fed6a85dd)
![Screenshot 2024-04-25 at 20 28 16](https://github.com/vscode-neovim/vscode-neovim/assets/45234439/4ce65d21-18e8-4802-8b2b-c06f0ded9fa8)
![Screenshot 2024-04-25 at 20 32 45](https://github.com/vscode-neovim/vscode-neovim/assets/45234439/121b0594-a48a-42e0-9132-032e11c5a200)
![Screenshot 2024-04-25 at 20 34 20](https://github.com/vscode-neovim/vscode-neovim/assets/45234439/e6e5e9ce-cb9c-490a-84fd-a732efde8b17)

So... why this doesn't work as key mapping? The if/else for plugins I do not have tested right now. Any recommendations?

**Comment by MannyFay** [comment]
By now, I have translated the vim script stuff from the documentation to Lua (there are plenty of examples :'D ...but I didn't see them in Lua like in my examples above).
Still, nothing works.

I've installed now the regular Vim extension in VS Code and deleted the Neovim extension. The Vim extension is way better. Way better written documentation, setting it up is super straight forward and easy. I've achieved with the Vim extension in two hours what I've not achieved with the Neovim extension in 2.5 days.

So, if I like to play around and enjoy wasting a lot of time, I choose the Neovim extension. If I have to get work done fast and seriously I use the Vim extension.

Regularly I will stay with my Neovim setup in terminal. VS Code is only for collaborations with Live Share, but I need my Neovim/Vim controls to be fast. So in my case, there is no time to have a childish play around with this Neovim extension.

**Reply by justinmk** [reply]
You are comparing apples to oranges. The "regular Vim extension" doesn't support vimscript or any of your plugins. If you want to skip your Nvim config in vscode-neovim, just set `vscode-neovim.neovimClean` as mentioned in the readme.

Though I agree that "childish" is a good way to describe this interaction.

**Reply by MannyFay** [reply]
I don't need. By now, I can handle all plugin functions I need with VS Codes built in actions and configuration of my Vim commands in settings.json together with them. You see, straight forward -> no playground. Me problems are fixed fast and easy by the Vim extension - and in my opinion, that is everything that counts.

uhuuu... let's get into the personal level ;)
-> Though I agree that "childish" is a good way to describe this interaction.
